Lib/posixpath.py

```diff
@@ -559,12 +560,11 @@ def commonpath(paths):
         curdir = '.'
 
     try:
+        prefix = min(roots)
```

Comment:
```suggestion
        prefix = roots[0]
```

Reply:
I believe it crashes faster:

```none
Traceback (most recent call last):
  File "c:\Users\wanne\Desktop\test.py", line 114, in <module>
    commonpath2(["foo"])
  File "c:\Users\wanne\Desktop\test.py", line 90, in commonpath2
    roots.sort()
    ^^^^^^^^^^
AttributeError: 'tuple' object has no attribute 'sort'
```

Jokes aside, here's a benchmark with `sorted()`:

```none
python -m timeit -s "import test" "test.commonpath1(('/foo', '//foo') * 100)"
python -m timeit -s "import test" "test.commonpath2(('/foo', '//foo') * 100)"
2000 loops, best of 5: 145 usec per loop # min (& any)
2000 loops, best of 5: 150 usec per loop # sort & slicing
# -> 1.03x slower
```

It isn't faster (rather slower) and sets off alarm bells: "O(NLOGN) > O(N)".
The current code is also more readable: get the minimum root, if there's any root.